src/Sarif.Viewer.VisualStudio.Core/RunningDocTableEventsHandler.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+using EnvDTE;
+
+using EnvDTE80;
+
+using Microsoft.Sarif.Viewer.ResultSources.Domain.Models;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Workspace.Indexing;
+
+namespace Sarif.Viewer.VisualStudio.Core
+{
+    /// <summary>
+    /// The class handles and listens to files being opened and fires events.
```

Comment:
add a bit more to summary -> for what purpose does it do the listening?  I presume `foobar` class takes a dependency on events from here?